caiman/base/movies.py

```diff
@@ -2202,13 +2202,42 @@ def load_iter(file_name, subindices=None, var_name_hdf5: str = 'mov', outtype=np
     if os.path.exists(file_name):
         extension = os.path.splitext(file_name)[1].lower()
         if extension in ('.tif', '.tiff', '.btf'):
-            Y = tifffile.TiffFile(file_name).pages
-            if subindices is not None:
-                if isinstance(subindices, range):
-                    subindices = slice(subindices.start, subindices.stop, subindices.step)
-                Y = Y[subindices]
-            for y in Y:
-                yield y.asarray().astype(outtype)
+            Y = tifffile.TiffFile(file_name).series[0]
+            dims = Y.shape[1:]
+            if len(dims)==3: # volumetric 3D data w/ multiple volumes per file
+                vol = np.empty(dims, dtype=outtype)
+                i = 0  # plane counter
+                if subindices is not None:
+                    if isinstance(subindices, slice):
+                        subindices = range(subindices.start,
+                        len(Y) if subindices.stop is None else subindices.stop,
+                        1 if subindices.step is None else subindices.step)
+                    t = 0  # volume counter
+                    for y in Y:
+                        if t in subindices:
+                            vol[i] = y.asarray()
+                        i += 1
+                        if i == dims[0]:
+                            i = 0
+                            if t in subindices:
+                                yield vol
+                            t +=1
+                else:
+                    for y in Y:
+                        vol[i] = y.asarray()
+                        i += 1
+                        if i == dims[0]:
+                            i = 0
+                            yield vol
+            elif len(dims)<3 and is3D: # volumetric 3D data w/ 1 volume per file
```

Comment:
Minor formatting thing: `len(dims)<3` might look better as `len(dims) < 3`

Reply:
changed it